README.md

```diff
@@ -3,3 +3,28 @@
 👻🔭
 
 https://jotai.org/docs/integrations/scope
+
+## Pro Tips
+
+1. Within a `ScopeProvider`, although an atom may not be scoped, its `atom.read` function could be called multiple times. **Therefore, do not use `atom.read` to perform side effects.**
+
+    NOTE: Async atoms always have side effects. To handle it, add additional code to prevent extra side effects. You can check this [issue](https://github.com/jotaijs/jotai-scope/issues/25#issuecomment-2014498893) as an example.
+
+2. Scoping a derived atom has no behavioral effect. Avoid it.
+
+    Derived atoms are merely proxies of primitive atoms. When scoping a derived atom, although the atom itself is copied, all of those copies still reference the same primitive atom, so their values are still shared.
+
+    It is a common pattern in Jotai to wrap the primitive atom within a utility function. To scope it properly, you should expose the primitive atom and then apply the scoping to it.
```

Comment:
I don't think we should encourage exposing base atoms. They are an implementation detail of the custom atom. Most developers use custom atoms from libraries out of the box. How would I scope atoms like `atomWithMachine`?